### DISCUSSION TITLE
Code structure

### DISCUSSION THREAD

**Discussion by johnryantaylor** [OceanBioME/OceanBioME.jl#18] [Ideas]
A few thoughts on the code structure:
- I think that it might make sense to keep the parameter list inside the biogeochemical model script (i.e. move the contents of params/lobster.jl into LOBSTER.jl).  Anyone modifying LOBSTER.jl will likely want to modify the default parameters too, so keeping them in the same place might be more convenient and would simplify the directory structure.
- We should probably organize the src folder a bit.  e.g. I would suggest a folder for BGC models.  Maybe also a folder for post processing which could contain plotting scripts?  Others will probably become clear as we go along.
- A very small things: can we change the name of the "example" folder to "examples"?  I don't know how to do this in github
- It would be nice to add an example that uses an analytical function instead of relying on the subpolar .nc files.  I like keeping the .nc files as an example for how to read and use data.

**Comment by jagoosw** [comment]
Hi John,

For the first two points I've essentially made these changes in the boundaries branch and am just finishing testing before I pull them in. 

I can have a go at the analytical one, and change the file name, tomorrow too.

Jago
